All_Relationships/1a.stratify-candidates.py

```diff
@@ -0,0 +1,189 @@
+DiseaseGene = candidate_subclass('DiseaseGene', ['Disease', 'Gene'])
+
+
+# # Make Stratified File
+
+# In[ ]:
+
+
+disease_ontology_df = pd.read_csv('https://raw.githubusercontent.com/dhimmel/disease-ontology/052ffcc960f5897a0575f5feff904ca84b7d2c1d/data/xrefs-prop-slim.tsv', sep="\t")
+disease_ontology_df = disease_ontology_df.drop_duplicates(["doid_code", "doid_name"])
+
+
+# In[ ]:
+
+
+gene_entrez_df = pd.read_csv('https://raw.githubusercontent.com/dhimmel/entrez-gene/a7362748a34211e5df6f2d185bb3246279760546/data/genes-human.tsv', sep="\t")
+gene_entrez_df = gene_entrez_df[["GeneID", "Symbol"]]
+
+
+# ## Map Each Disease to Each Gene
+
+# In[ ]:
+
+
+gene_entrez_df['dummy_key'] =0
+disease_ontology_df['dummy_key'] = 0
+dg_map_df = gene_entrez_df.merge(disease_ontology_df[["doid_code", "doid_name", "dummy_key"]], on='dummy_key')
+
+
+# ## Label All Pairs Whether or Not They are in Hetnets
+
+# In[ ]:
+
+
+get_ipython().run_cell_magic(u'time', u'', u'hetnet_kb_df = pd.read_csv("hetnet_dg_kb.csv")\nhetnet_set = set(map(lambda x: tuple(x), hetnet_kb_df.values))\nhetnet_labels = np.ones(dg_map_df.shape[0]) * -1\n\nfor index, row in tqdm.tqdm(dg_map_df.iterrows()):\n    if (row["doid_code"], row["GeneID"]) in hetnet_set:\n        hetnet_labels[index] = 1 \n    \ndg_map_df["hetnet"] = hetnet_labels')
+
+
+# ## See if D-G Pair is in Pubmed
+
+# In[ ]:
+
+
+get_ipython().run_cell_magic(u'time', u'', u'pubmed_dg_pairs = set({})\ncands = []\nchunk_size = 1e5\noffset = 0\n\nwhile True:\n    cands = session.query(DiseaseGene).limit(chunk_size).offset(offset).all()\n    \n    if not cands:\n        break\n        \n    for candidate in tqdm.tqdm(cands):\n        pubmed_dg_pairs.add((candidate.Disease_cid, candidate.Gene_cid))\n    \n    offset = offset + chunk_size')
+
+
+# In[ ]:
+
+
+pubmed_labels = np.ones(dg_map_df.shape[0]) * -1
+
+for index, row in tqdm.tqdm(dg_map_df.iterrows()):
+    if (row["doid_code"], str(row["GeneID"])) in pubmed_dg_pairs:
+        pubmed_labels[index] = 1
+
+dg_map_df["pubmed"] = pubmed_labels
+
+
+# In[ ]:
+
+
+dg_map_df = dg_map_df.rename(index=str, columns={"GeneID": "gene_id", "doid_code": "disease_id", "doid_name": "disease_name", "Symbol":"gene_name"})
+dg_map_df["hetnet"] = dg_map_df["hetnet"].astype(int)
+dg_map_df["pubmed"] = dg_map_df["pubmed"].astype(int)
+dg_map_df.to_csv("dg_map.csv", index=False)
+
+
+# ## Modify the Candidate split
+
+# This code below changes the split column of the candidate table as mentioned above. Using sqlalchemy and the chunking strategy, every candidate that has the particular disease entity id (DOID:3393) will be given the category of 2. 2 Representes the testing set which will be used in the rest of the notebooks.
+
+# In[5]:
+
+
+dg_map_df = pd.read_csv("dg_map.csv")
+
+
+# In[6]:
+
+
+print dg_map_df[(dg_map_df["hetnet"] == 1)].shape
+print dg_map_df[(dg_map_df["pubmed"]== 1)].shape
+print
+print dg_map_df[(dg_map_df["hetnet"] == 1)&(dg_map_df["pubmed"]== 1)].shape
```

Comment:
You may want to start putting parenthesis arround print statements to make the python 3 upgrade easier